### PR TITLE
Add diffusion-implicit time stepping heat tutorial

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR adds a diffusion-implicit heat tutorial. cc @ChrisRackauckas

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,6 +64,7 @@ SciMLTutorials.open_notebooks()
 - Advanced
   - [A 2D Cardiac Electrophysiology Model (CUDA-accelerated PDE solver)](http://tutorials.juliadiffeq.org/html/advanced/01-beeler_reuter.html)
   - [Solving Stiff Equations](http://tutorials.juliadiffeq.org/html/advanced/02-advanced_ODE_solving.html)
+  - [Solving Stiff Equations (the heat equation)](http://tutorials.juliadiffeq.org/html/advanced/04-diffusion_implicit_heat_equation.html)
```
Comment (ChrisRackauckas) [inline README.md:67]:
```suggestion
  - [Solving the heat equation with diffusion-implicit time-stepping](http://tutorials.juliadiffeq.org/html/advanced/04-diffusion_implicit_heat_equation.html)
```

**Comment by ChrisRackauckas** [comment]
Interestingly, Buildkite isn't firing. Let's see what happens.

**Comment by ChrisRackauckas** [comment]
https://tutorials.sciml.ai/html/advanced/04-diffusion_implicit_heat_equation.html 🎉 and the .md will hopefully get docs integration soon enough.

**Comment by charleskawczynski** [comment]
> https://tutorials.sciml.ai/html/advanced/04-diffusion_implicit_heat_equation.html 🎉 and the .md will hopefully get docs integration soon enough.

Excellent, thanks!

**Pull request merged**